tests/manage/test_disruptive.py

```diff
@@ -0,0 +1,134 @@
+        operations_instance (Class instance): The instance of the class
+        operation (str): The name of the operation to look for
+
+    Returns:
+        tuple: instance method object along with the method name
+    """
+    operations_pool = [i for i in inspect.getmembers(operations_instance, predicate=inspect.ismethod)]
+    ret = [i for i in operations_pool if i[0] == operation][0]
+    return ret
+
+
+class BaseDisruption(ManageTest):
+    """
+    Base class for PVC related disruption tests
+    """
+    def disruptive_base(self, operation_to_disrupt, harm, resource):
+        """
+        Base function for PVC disruptive tests
+        """
+        dis_op = get_operation_obj(operations_instance=DISRUPTION_OPS, operation=f"{harm}_{resource}")
+        get_operation_obj(operations_instance=DISRUPTION_OPS, operation=f"get_{resource}")[1]()
+        pvc_obj = helpers.create_pvc(
+            sc_name=self.sc_obj.name, wait=False
+        )
+        if operation_to_disrupt == 'create_pvc':
+            dis_op[1]()
+        pvc_obj.reload()
+        assert pvc_obj.ocp.wait_for_resource(condition=constants.STATUS_BOUND, resource_name=pvc_obj.name)
+
+        pod_obj = helpers.create_pod(
+            interface_type=constants.CEPHBLOCKPOOL, pvc=pvc_obj.name, wait=False
+        )
+        if operation_to_disrupt == 'create_pod':
+            dis_op[1]()
+        pod_obj.reload()
+        assert pod_obj.ocp.wait_for_resource(condition=constants.STATUS_RUNNING, resource_name=pod_obj.name)
+
+
+@pytest.mark.usefixtures(
+    create_rbd_secret.__name__,
+    create_ceph_block_pool.__name__,
+    create_rbd_storageclass.__name__,
+)
+class TestRBDDisruption(BaseDisruption):
+    """
+    RBD PVC related disruption tests class
+    """
+    # TODO: for each test iteration, add Polarion ID
+    @pytest.mark.parametrize(
+        argnames=["operation_to_disrupt", "harm", "resource"],
+        argvalues=[
+            pytest.param(*['create_pvc', 'delete', 'mgr']),
+            pytest.param(*['create_pod', 'delete', 'mgr']),
+            pytest.param(*['create_pvc', 'delete', 'mon']),
+            pytest.param(*['create_pod', 'delete', 'mon']),
+            pytest.param(*['create_pvc', 'delete', 'osd']),
+            pytest.param(*['create_pod', 'delete', 'osd']),
+            # pytest.param(*['create_pvc', 'delete_disk', 'mgr']),
+            # pytest.param(*['create_pod', 'delete_disk', 'mgr']),
+            # pytest.param(*['create_pvc', 'delete_disk', 'mon']),
+            # pytest.param(*['create_pod', 'delete_disk', 'mon']),
+            # pytest.param(*['create_pvc', 'delete_disk', 'osd']),
+            # pytest.param(*['create_pod', 'delete_disk', 'osd']),
+        ]
+    )
+    @tier4
+    def test_disruptive_block(self, operation_to_disrupt, harm, resource):
+        """
+        RBD PVC related disruption tests class method
+        """
+        self.disruptive_base(operation_to_disrupt, harm, resource)
+
+
+# @pytest.mark.usefixtures(
```

Comment:
Commented unused code should be removed.

Reply:
@vasukulkarni done